test/inductor/test_torchinductor.py

```diff
@@ -4217,15 +4217,15 @@ def fn(a, b):
             ),
         )
 
-    @xfail_if_mps
     def test_device_assert(self):
         def fn(x, y):
             x = torch.sum(x.view(int(x.shape[0] / 6), 6), dim=1)
             return torch.gather(x, 0, torch.trunc(y).to(torch.int64))
 
         x1 = torch.randn(30, device=self.device)
         x2 = torch.randn(36, device=self.device)
-        y = torch.ones(1, dtype=torch.float64, device=self.device)
+        dtype = torch.float64 if self.device != "mps" else torch.float32
```

Comment:
instead of saying `self.device != mps` can we say if the dtype is supported?

Reply:
Good followup PR, as I have this anti-pattern all over the place...